apps/remix-ide/src/app/files/dgitProvider.js

```diff
@@ -105,15 +105,38 @@ class DGitProvider extends Plugin {
   }
 
   async checkout (cmd, refresh = true) {
+    const gitmodules = await this.parseGitmodules() || []
     await git.checkout({
       ...await this.getGitConfig(),
       ...cmd
     })
-    if (refresh) {
+    const newgitmodules = await this.parseGitmodules() || []
+    // find the difference between the two gitmodule versions
+    const toRemove = gitmodules.filter((module) => {
+      return !newgitmodules.find((newmodule) => {
+        return newmodule.name === module.name
+      })
+    })
+
+    for (const module of toRemove) {
```

Comment:
You also need to clone the git module that are not already here right? or this done elsewhere.

Reply:
somewhere else